tests/test_core.py

```diff
@@ -267,27 +267,53 @@ def test_magphase():
     assert np.allclose(S * P, D)
 
 
-def test_istft():
-    def __test(infile):
-        DATA = load(infile)
+def test_istft_reconstruction():
+    from scipy.signal import bartlett, hann, hamming, blackman, blackmanharris
 
-        if DATA['hann_w'][0, 0] == 0:
-            window = np.ones
-            win_length = 2 * (DATA['D'].shape[0] - 1)
-        else:
-            window = None
-            win_length = DATA['hann_w'][0, 0]
+    def __test(x, n_fft, hop_length, window, atol):
+        S = librosa.core.stft(
+            x, n_fft=n_fft, hop_length=hop_length, window=window)
+        x_reconstructed = librosa.core.istft(
+            S, hop_length=hop_length, window=window)
 
-        Dinv = librosa.istft(DATA['D'],
-                             hop_length=DATA['hop_length'][0, 0].astype(int),
-                             win_length=win_length,
-                             window=window,
-                             center=False)
+        L = min(len(x), len(x_reconstructed))
+        x = np.resize(x, L)
+        x_reconstructed = np.resize(x_reconstructed, L)
 
-        assert np.allclose(Dinv, DATA['Dinv'])
+        # NaN/Inf/-Inf should not happen
+        assert np.all(np.isfinite(x_reconstructed))
 
-    for infile in files('data/core-istft-*.mat'):
-        yield (__test, infile)
+        # should be almost approximately reconstucted
+        assert np.allclose(x, x_reconstructed, atol=atol)
+
+    # White noise
+    np.random.seed(98765)
+    x1 = np.random.randn(2**15)
+
+    # Sin wave
+    x2 = np.sin(np.linspace(-np.pi, np.pi, 2**15))
+
+    # Real music signal
+    x3, sr = librosa.load('data/test1_44100.wav', sr=None, mono=True)
+    assert sr == 44100
+
+    for x, atol in [(x1, 1.0e-6), (x2, 1.0e-7), (x3, 1.0e-7)]:
+        for window_func in [bartlett, hann, hamming, blackman, blackmanharris]:
+            for n_fft in [512, 1024, 2048, 4096]:
+                win = window_func(n_fft, sym=False)
+                symwin = window_func(n_fft, sym=True)
+                for hop_length_denom in six.moves.range(2, 10):
+                    hop_length = n_fft // hop_length_denom
+                    yield (__test, x, n_fft, hop_length, win, atol)
+                    yield (__test, x, n_fft, hop_length, symwin, atol)
+
```

Comment:
For full test coverage, there should also be a test that passes ``window_func`` directly as a function to ``stft`` and ``istft``, rather than the pre-computed windows we have currently.